internal/services/elasticsan/elastic_san_volume_snapshot_data_source.go

```diff
@@ -0,0 +1,147 @@
+package elasticsan
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/elasticsan/2023-01-01/snapshots"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/elasticsan/2023-01-01/volumes"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/elasticsan/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type ElasticSANVolumeSnapshotDataSource struct{}
+
+var _ sdk.DataSource = ElasticSANVolumeSnapshotDataSource{}
+
+type ElasticSANVolumeSnapshotDataSourceModel struct {
+	Name                string                                 `tfschema:"name"`
+	SourceVolumeSizeGiB int64                                  `tfschema:"source_volume_size_gib"`
```

Comment:
In the other elastic san resources we're using `something_in_gib`, can we rename this to follow that pattern as well 
```suggestion
	SourceVolumeSizeGiB int64                                  `tfschema:"source_volume_size_in_gib"`
```